### PR TITLE
Added api contract test case for logs endpoint

### PR DESCRIPTION
**Pull request by Kashatlast2**
## Which Traffic Control components are affected by this PR?
- Traffic Ops

## What is the best way to verify this PR?
Step 1 : Follow steps from README file(traffic_ops/testing/api_contract) to run api-contract-tests.

Step 2 : To execute API Contract Testcase , navigate to traffic_ops/testing/api_contract/v4
pytest --to-user Username --to-password Password --to-url URL --config Config Path --request-template Request template --response-template Response template

Note: Config, Request template and response template paths are optional.

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 1]
```diff
@@ -2160,5 +2160,36 @@
...
+            "ticketNum",
+            "level",
+            "lastUpdated",
+            "user",
+            "id",
+            "message"
+        ],
+        "properties": {
+            "ticketNum": {
+                "type": "null"
```
Comment (ericholguin) [inline traffic_ops/testing/api_contract/v4/data/response_template.json:2176]: this should be an integer

Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: Updated to integer.

File: traffic_ops/testing/api_contract/v4/test_logs.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	for log in logs_data:
+		if isinstance(log, dict):
+			change_log_id = log.get("id")
+			if not isinstance(change_log_id, int):
+				raise TypeError("Malformed log in prerequisite data; 'id' not an integer")
+
+			# Hitting logs GET response
+			change_logs_get_response: tuple[JSONData, requests.Response] = to_session.get_change_logs()
+
+			logs_data, _ = change_logs_get_response
```
Comment (TerekhovaKate) [inline traffic_ops/testing/api_contract/v4/test_logs.py:53]: check the variable declaration...

Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/test_logs.py:53]: change_logs_data variable changed to logs_data

**Review by ericholguin — CHANGES_REQUESTED** [review]

File: traffic_ops/testing/api_contract/v4/conftest.py [hunk 3]
```diff
@@ -1945,3 +1945,23 @@ def server_server_capabilities_data_post(to_session: TOSession, edge_type_data:N
...
+    PyTest Fixture to retrieve log data from logs endpoint.
+    :param to_session: Fixture to get Traffic Ops session.
+    :returns: Log data obtained from logs endpoint.
+    """
+
+	change_logs = check_template_data(request_template_data["logs"], "logs")
+
+
+    # Hitting logs GET methed
+	response: tuple[JSONData, requests.Response] = to_session.get_change_logs(data=change_logs)
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/conftest.py:1963]: There is no POST Request available. And when I tried to create it. It was telling me that the endpoint doesn't exist. Same for in Postman.

I was getting 

`self = <trafficops.tosession.TOSession object at 0x107365790>, operation = 'post', api_path = 'logs', query_params = None, munchify = True
debug_response = False, expected_status_codes = range(200, 300), unused_args = ()
kwargs = {'data': {'id': 444, 'level': 'APICHANGE', 'message': 'log for test purposes', 'ticketNum': None, ...}}
msg = 'POST request to RESTful API at [https://tp.staging.cdnlab.comcast.net:443/api/4.0/logs] expected status(s) range(200,...nd; Response: {\'alerts\': [{\'text\': "The requested path \'/api/4.0/logs\' does not exist.", \'level\': \'error\'}]}'
response = <Response [404]>, retdata = {'alerts': [{'level': 'error', 'text': "The requested path '/api/4.0/logs' does not exist."}]}
endpoint = 'https://tp.staging.cdnlab.comcast.net:443/api/4.0/logs'
params = {'data': '{"ticketNum": null, "level": "APICHANGE", "user": "admin", "id": 444, "message": "log for test purposes"}', 'headers': {'Content-Type': 'application/json; charset=UTF-8'}, 'verify': False}`

GET to the same endpoint works fine so I thought I had to modify for GET for this one as it wasn't accepting POST requests.

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 4]
```diff
@@ -2160,5 +2160,39 @@
...
+        ],
+        "properties": {
+            "ticketNum": {
+                "type": [
+                    "integer",
+                    "null"
+                ]
+            },
+            "level": {
+                "type": "string"
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: added null

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 5]
```diff
@@ -2160,5 +2160,39 @@
...
+            },
+            "lastUpdated": {
+                "type": "string"
+            },
+            "user": {
+                "type": "string"
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: added null

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 6]
```diff
@@ -2160,5 +2160,39 @@
...
+            },
+            "id": {
+                "type": "integer"
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: added null

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 7]
```diff
@@ -2160,5 +2160,39 @@
...
+            },
+            "message": {
+                "type": "string"
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: added null

File: traffic_ops/testing/api_contract/v4/data/response_template.json [hunk 8]
```diff
@@ -2160,5 +2160,39 @@
...
+                "type": [
+                    "integer",
+                    "null"
+                ]
+            },
+            "level": {
+                "type": "string"
+            },
+            "lastUpdated": {
+                "type": "string"
```
[the remark this answers is not in the document]
Comment (Kashatlast2) [reply traffic_ops/testing/api_contract/v4/data/response_template.json]: added null

**Pull request merged**